### ISSUE TITLE
Multiple server compatibility

### ISSUE THREAD

**Issue by Nzii3** [kyb3r/modmail#3245]
### Is your feature request related to a problem? Please elaborate.

Please add the ability to have the modmail system worn across multiple servers. This could be achieved with a select menu that is sent (if multiple guild IDs are passed in the `.env` file) with a list of all the guilds by name, when a server is selected through the menu, it will fetch that guild as the modmail guild, then proceed with creating the thread as usual. The user would only be able to have 1 thread open at a time between all the servers, for obvious reasons.

A dictionary could be stored in the database with a guild ID linking to a modmail category ID, for example:
```json
# guild ID          category ID
{1234567890: 0987654321}
```

### Describe the solution you'd like

This would allow us to use multiple modmail servers, so you can share one modmail system with other users.

### Does your solution involve any of the following?

- [ ] Logviewer
- [X] New config option

### Describe alternatives you've considered

Using one modmail server, but this isn’t an ideal solution because I’d like to have multiple servers that threads can be sent to.

### Who will this benefit?

Users who want to use the bot across multiple servers with multiple modmail servers. This would decrease the hassle of having to create a new kyber modmail bot for **each** server we want to use this great system on.

### Additional Information

*I’ve created this type of thing in my own modmail bots, and I would like to see it in a widely-used and known modmail system.*

**Comment by sebkuip** [comment]
This would literally require a complete rewrite of the entire bot. Best case this would take months, but more realistically seeing how often people are busy irl this would maybe take more like a year or 2 to rewrite and have this function.

Just set up a second instance of modmail for your other server.

**Comment by Nzii3** [comment]
This feature would remove the need to create another instance, which is really annoying and complicated. I‘ll gladly make PRs but I’m sure many other people would like to see this implemented. It makes the bot usable in almost as many servers as you want. Think about it, you can have one bot named “Support Bot” (or something) and use it in multiple servers with multiple modmail servers. 

The entire code wouldn’t have to be rewritten, only the basic stuff of the thread handler and how a thread is created. The guild ID where the thread was actually created, can be stored in the database. Then when the thread is fetched, you have the guild ID and the thread channel ID.

**Comment by Ushie** [comment]
> The guild ID where the thread was actually created

What thread? unless there's a different system than the one I personally use on ModMail, people DM the bot to start a thread, there will be no guild ID involved

**Comment by Nzii3** [comment]
> > The guild ID where the thread was actually created
> 
> What thread? unless there's a different system than the one I personally use on ModMail, people DM the bot to start a thread, there will be no guild ID involved

The bot has to fetch a channel object and that channel object has an attribute `guild`. That’s what I’m talking about. The thread channel that staff sees.

**Comment by Ushie** [comment]
> > > The guild ID where the thread was actually created
> > 
> > 
> > What thread? unless there's a different system than the one I personally use on ModMail, people DM the bot to start a thread, there will be no guild ID involved
> 
> The bot has to fetch a channel object and that channel object has an attribute `guild`. That’s what I’m talking about. The thread channel that staff sees.

My apologies, I missed the dropdown menu part in the first message, I'll mark my messages as resolved 👍

**Comment by Nzii3** [comment]
Update?

**Comment by martinbndr** [comment]
> Update?

@Nzii3 Nope. As sebkuip already told, it would require a nearly whole rewrite of the bot which would take much time. The bot is not designed for working with multiple guilds as inbox servers like you explained using dropdown etc. I guess this won´t ever be released.
Just host another bot instance would be the solution.

**Comment by StephenDaDev** [comment]
This is also possible to achieve through the upcoming `threadmenu` system, given that all threads will still go to a single inbox server that is common regardless of server initiated from. A key selling point of this modmail over its competitors is the ability to send a single DM and immediately be in contact with the server's team. With everything listed by me and my colleagues, I will be closing this as "will not implement".

Thank you for providing your suggestions for the bot.
